### PR TITLE
Remove eslint configs and plugins that come with eslint-config-next from direct dependencies

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Simpler fix for eslint import order inconsistencies:
- remove direct dependencies on eslint components that come with eslint-config-next to remove risk of version conflict
- rename base config file for better editing experience
- rollback lint-staged changes

## Justify why they are needed

Better developer experience, fixable eslint errors are fixed and comitted automatically

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1372** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1372" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1372?utm_source=stack-comment).

**Review by KarnellSchultz — APPROVED** [review]
Works for me 👍 

parks-and-recreation-win-gif-by-hulu

**Review by gustaveen — APPROVED** [review]
lgtm! 🙌

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
```
Comment (robinandeer) [inline .eslintrc.js:1]: What do we need this for actually? We don't have any files outside of the workspaces that we want to lint, no?

**Review by robinandeer — APPROVED** [review]
Aha! Nice find :D

Didn't realize these came with next config. And I suppose it doesn't hurt to use the next config for UI package either.

**Pull request merged**